src/V3MergeCond.cpp

```diff
@@ -227,17 +605,18 @@ class MergeCondVisitor final : public VNVisitor {
                 return condTrue ? maskLsb(andp->lhsp()->unlinkFrBack())
                                 : new AstConst{rhsp->fileline(), AstConst::BitFalse()};
             }
-        } else if (VN_IS(rhsp, WordSel) || VN_IS(rhsp, VarRef) || VN_IS(rhsp, Const)) {
+        } else if (VN_IS(rhsp, ArraySel) || VN_IS(rhsp, WordSel) || VN_IS(rhsp, VarRef)
+                   || VN_IS(rhsp, Const)) {
             return rhsp->cloneTree(false);
         }
+        // LCOV_EXCL_START
         rhsp->dumpTree("Don't know how to fold expression: ");
-        rhsp->v3fatalSrc("Don't know how to fold expression");
+        rhsp->v3fatalSrc("Should not try to fold this during conditional merging");
```

Comment:
In dumpTree line above put "if (debug())" in front.  We shouldn't dump trees generally even on assertions as will confuse users even more.